tests/unittests/Foundation/NSUndoManagerTests.mm

```diff
@@ -0,0 +1,718 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+
+@interface UndoTestObject : NSObject
+@property (retain) NSUndoManager* undoManager;
+@property (retain) NSNumber* value;
+@property (retain) NSMutableArray* arrayOfNumbers;
+- (id)initWithValue:(NSNumber*)num;
+- (id)initWithEmptyArray;
+- (void)updateNumber:(NSNumber*)value;
+- (void)updateNumberSingleUndoStep:(NSNumber*)value;
+- (void)addToMutableArray:(NSNumber*)value;
+- (void)removeFromMutableArray:(NSNumber*)value;
+@end
+
+@implementation UndoTestObject
+
+- (id)initWithValue:(NSNumber*)num {
+    self = [super init];
+    _undoManager = [[NSUndoManager alloc] init];
+    _value = [num retain];
+    _arrayOfNumbers = nil;
+    return self;
+}
+
+- (id)initWithEmptyArray {
+    self = [super init];
+    _undoManager = [[NSUndoManager alloc] init];
+    _value = nil;
+    _arrayOfNumbers = [[NSMutableArray array] retain];
+    return self;
+}
+
+- (void)dealloc {
+    [_undoManager release];
+    [_value release];
+    [_arrayOfNumbers release];
+}
```

Comment:
Can you search for other deallocs in this code review and make sure they call `super` too?

---
In reply to: [77097364](https://github.com/Microsoft/WinObjC/pull/900#discussion_r77097364) [](ancestors = 77097364)